File: src/bubble_sort/bubblesort.py
def bubble_sort(a):
    is_sorted = False
    len_a = len(a)

    while not is_sorted:
        is_sorted = True

        for i in range(len_a):
            current = a[i]

            if i != len_a - 1 and current > a[i + 1]:
                a[i] = a[i + 1]
                a[i + 1] = current
                is_sorted = False

File: src/bubble_sort/bubbleSort.ts
export function bubbleSort(arr: number[]): number[] {
  let isSorted = false;

  while (!isSorted) {
    isSorted = true;

    for (let i = 0; i < arr.length; i++) {
      const current = arr[i];
      const isLast = i === arr.length - 1;

      if (!isLast && arr[i] > arr[i + 1]) {
        arr[i] = arr[i + 1];
        arr[i + 1] = current;
        isSorted = false;
      }
    }
  }

  return arr;
}
